processor/src/test/java/org/mapstruct/ap/test/selection/qualifier/iterable/TopologyMapper.java

```diff
@@ -29,6 +32,14 @@
     @Mapping( target = "topologyFeatures", qualifiedBy = Cities.class )
     public abstract TopologyEntity mapTopologyAsCity(TopologyDto dto);
 
+    @Rivers
+    @IterableMapping( qualifiedBy = Rivers.class )
+    public abstract List<TopologyFeatureEntity> mapTopologiesAsRiver(List<TopologyFeatureDto> in);
```

Comment:
What was the reason for adding this 2 abstract methods? Are these methods failing without your changes?

Reply:
the qualifier acts both on the element as on the `@BeforeMapping` now qualifiers are working as they should, MapStruct cannot find a qualifying method anymore for the element.

Reply:
So this means that this is a backwards incompatible change? In this case I am not quite sure if it is OK to break it. Although you can say that if you are adding the qualifiedBy we should report thay we didn't find that

Reply:
as discussed elsewhere..